sdk/core/Azure.Core/src/Shared/ConnectionString.cs

```diff
@@ -37,8 +37,16 @@ private ConnectionString(Dictionary<string, string> pairs, string pairSeparator,
         public string? GetNonRequired(string keyword) =>
             _pairs.TryGetValue(keyword, out var value) ? value : null;
 
-        public bool TryGetSegmentValue(string keyword, out string? value) =>
-            _pairs.TryGetValue(keyword, out value);
+        public bool TryGetSegmentValue(string keyword, out string? value, string? defaultValue = null)
```

Comment:
Is there a test in `Azure.Core` that we can extend to cover the new functionality - or is there an existing fixture that we can add to?

Reply:
Good callout - I'll add some coverage.

Reply:
Let's add another method instead. 

`GetSegmentValueOrDefault()`

When using the default value the return value is always `true` and the method is just more awkward to use.

``` C#
var defauledValue = CS.GetSegmentValueOrDefault("Segment", "Default");
//vs
CS.TryGetSegmentValue("Segment", out defauledValue , "Default");
```

Reply:
That's fair - I was being lazy :)